### PR TITLE
Bug-fix: Roberta Embeddings Not Masked

### LINKED ISSUE #1761: Roberta Positional Embeddings Not Masked
## Bug

Hi, I think for the RoBERTa model, the positional embeddings are created slightly wrong. In both this library and in `fairseq` the positional embeddings are sequential integers starting from `padding_idx + 1`. However, in `fairseq.utils.make_positions` all indicies of the input that correspond to the padding index are masked as they go into the positional embedding layer.

## Example

```python
import torch
padding_idx = 0

tensor = torch.as_tensor([
    [1, 13, 54, 0, 0],
    [1, 55, 12, 24, 0]
])
```

### In Fairseq

```
from fairseq import utils
utils.make_positions(tensor, padding_idx = padding_idx)
>>> tensor([
        [1, 2, 3, 0, 0],
        [1, 2, 3, 4, 0]
        ])
```

The positional sequence begins at padding_idx + 1 but all occurrences of the padding index in the input tensor are unmanipulated. This means the position embeddings are only adjusted for tokens which are not padding.

### In Transformers

This code comes from [modeling_roberta.py#L65](https://github.com/huggingface/transformers/blob/master/transformers/modeling_roberta.py#L65)
```python
# Position numbers begin at padding_idx+1. Padding symbols are ignored.
# cf. fairseq's `utils.make_positions`
position_ids = torch.arange(self.padding_idx+1, seq_length+self.padding_idx+1, dtype=torch.long, device=input_ids.device)
position_ids = position_ids.unsqueeze(0).expand_as(input_shape)
```

The output tensor of these function calls do not retain the masked tokens:
```
>>> tensor([
        [1, 2, 3, 4, 5],
        [1, 2, 3, 4, 5]
        ])
```
## Complete Example

```python

from fairseq import utils

tensor = torch.as_tensor([
    [1, 13, 54, 0, 0],
    [1, 55, 12, 24, 0]
])

def fairseq_make_positions(tensor, padding_idx):
    mask = tensor.ne(padding_idx).int()
    return (
        torch.cumsum(mask, dim=1).type_as(mask) * mask
    ).long() + padding_idx

def transformers_make_positions(input_ids, padding_idx):
    input_shape = input_ids.size()
    seq_length = input_shape[1]
    position_ids = torch.arange(padding_idx+1, seq_length+padding_idx+1, dtype=torch.long, device=input_ids.device)
    position_ids = position_ids.unsqueeze(0).expand(input_shape)
    return position_ids


print(fairseq_make_positions(tensor, padding_idx = 0))
print(transformers_make_positions(tensor, padding_idx = 0))
```
Output:
```
# Fairseq:
tensor([[1, 2, 3, 0, 0],
        [1, 2, 3, 4, 0]])
# Transformers:
tensor([[1, 2, 3, 4, 5],
        [1, 2, 3, 4, 5]])
```

## Expected behavior
I expect all padded tokens in the input tensor to be padded as they go into the positional embeddings layer as well as the token embeddings layer.

## Additional context

Happy to make a PR if this is agreed to be a bug, but this will mean the positional embeddings train slightly differently to the existing `RobertaEmbeddings` module.

### PR DESCRIPTION
**Pull request by DomHudson**
## Summary
I replace the code that makes the position ids with logic closer to the original fairseq `make_positions` function. It wasn't clear to me what to do in the event that the embeddings are passed in directly through `inputs_embeds` so I resorted to the old methodology and just generating a positional id for all inputs.

## Closes
#1761

### CODE REVIEW

**Comment by julien-c** [comment]
Thanks! Any way you could do this PR with just the relevant changes (no whitespace changes, no extraneous commits)?

**Comment by DomHudson** [comment]
Hi,

I have pushed just the relevant changes with a single commit.

Many thanks,
Dom

**Comment by julien-c** [comment]
Thanks! We need to update the TF model as well. (this might be the reason behind the failing unit test, btw)

Will do it unless you have the bandwidth (and the inclination) to do it yourself @DomHudson

File: transformers/tests/modeling_roberta_test.py [hunk 1]
```diff
@@ -235,17 +239,24 @@ def test_create_position_ids_from_inputs_embeds(self):
...
+        ]
+        expected_positions = torch.as_tensor([expected_single_positions, expected_single_positions])
+        position_ids = embeddings.create_position_ids_from_inputs_embeds(inputs_embeds)
+        self.assertEqual(
+            position_ids.shape,
+            expected_positions.shape
+        )
+        self.assertTrue(
+            torch.all(torch.eq(position_ids, expected_positions))
+        )
```
Comment (julien-c) [inline transformers/tests/modeling_roberta_test.py]: this updated test was failing

**Comment by julien-c** [comment]
@DomHudson Fixed a bug introduced by this PR's commit + rebased on top of current master.

Please note that for most models, we expect users to define `attention_mask`s themselves. 

If I'm not mistaken, if a user defined an attention_mask that correctly excluded padding, he would not need the padding-aware position_ids from this PR.

**Comment by thomwolf** [comment]
LGTM merging

**Comment by DomHudson** [comment]
Hi @thomwolf @julien-c,
Apologies that I went awol, and thank you for merging the fix.

I can't claim to fully understand GitHub, how were my changes merged in (via commit f15f087) without me registering as the author? It looks like the merged changes are identical to my original PR. Was the merge performed off GitHub?
